src/AzureClient/EntryPoint/IEntryPoint.cs

```diff
@@ -26,5 +25,19 @@ public interface IEntryPoint
         /// <param name="cancellationToken">Cancellation token used to interrupt this submission.</param>
         /// <returns>The details of the submitted job.</returns>
         public Task<IQuantumMachineJob> SubmitAsync(IQuantumMachine machine, AzureSubmissionContext submissionContext, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Submits the entry point for execution to Azure Quantum.
+        /// </summary>
+        /// <param name="submitter">The <see cref="IQirSubmitter"/> object representing the job submission target.</param>
+        /// <param name="submissionContext">The <see cref="AzureSubmissionContext"/> object representing the submission context for the job.</param>
+        /// /// <param name="cancellationToken">Cancellation token used to interrupt this submission.</param>
+        /// <returns>The details of the submitted job.</returns>
+        public Task<IQuantumMachineJob> SubmitAsync(IQirSubmitter submitter, AzureSubmissionContext submissionContext, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// The stream from which QIR bitcode for the entry point can be read.
+        /// </summary>
+        public Stream? QirStream { get; }
```

Comment:
I am curious why `QirStream` needs to be part of this interface? Since `SubmitAsync` is implemented in the `EntryPoint` class it can access `QirStream` there without the need of including it in this interface too.

Reply:
The QirStream is used here in QirMagic.cs: https://github.com/microsoft/iqsharp/blob/36f2ebffe60bf66261435c4dad99e96d47c833f2/src/Kernel/Magic/QirMagic.cs#L92